src/python/WMCore/WorkQueue/WorkQueue.py

```diff
@@ -953,34 +961,40 @@ def performSyncAndCancelAction(self, skipWMBS):
 
                     updated_elements = [x for x in result['Elements'] if x.modified]
                     for x in updated_elements:
-                        self.logger.debug("Updating progress %s (%s): %s" % (x['RequestName'], x.id, x.statusMetrics()))
+                        self.logger.debug("Updating progress %s (%s): %s", x['RequestName'], x.id, x.statusMetrics())
                     for x in updated_elements:
                         self.backend.updateElements(x.id, **x.statusMetrics())
 
                 if not parentQueueDeleted:
-                    self.logger.info('Waiting for parent queue to delete "%s"' % wf)
+                    self.logger.info('Waiting for parent queue to delete "%s"', wf)
 
             except Exception as ex:
-                self.logger.error('Error processing workflow "%s": %s' % (wf, str(ex)))
+                self.logger.error('Error processing workflow "%s": %s', wf, str(ex))
 
         msg = 'Finished elements: %s\nCanceled workflows: %s' % (', '.join(["%s (%s)" % (x.id, x['RequestName']) \
                                                                             for x in finished_elements]),
                                                                  ', '.join(wf_to_cancel))
         self.backend.recordTaskActivity('housekeeping', msg)
 
     def performQueueCleanupActions(self, skipWMBS=False):
+        try:
+            # fetch workflows known to workqueue + workqueue_inbox and with spec attachments
+            wqRequests = self.backend.getWorkflows(includeInbox=True, includeSpecs=True)
```

Comment:
Apparently the list of requests can chance between this call and the execution of `performSyncAndCancelAction`, which raises the following error:
```
2019-05-09 23:51:13,511:140622703253248:ERROR:WorkQueue:Error processing workflow "amaltaro_ACDC_TC_PreMix_Agent122_Validation_190509_150523_6298": Can't get policy, no elements or parents
```

I'll add the double workqueue call back and see whether that error goes away. BTW, it only happens in a short time window, so I saw a few errors in my tests, not in every thread cycle though.